modules/sdk/src/main/scala/org/tessellation/sdk/domain/fork/ForkInfoEntries.scala

```diff
@@ -0,0 +1,40 @@
+package org.tessellation.sdk.domain.fork
+
+import cats.syntax.eq._
+import cats.{Eq, Show}
+
+import scala.collection.immutable.Queue
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosInt
+
+sealed trait ForkInfoEntries {
+  def add(entry: ForkInfo): ForkInfoEntries
+  def getEntries: Iterable[ForkInfo]
+}
+
+object ForkInfoEntries {
+  implicit val eq: Eq[ForkInfoEntries] = Eq.instance[ForkInfoEntries] { (e1, e2) =>
+    e1.getEntries.toSeq === e2.getEntries.toSeq
+  }
+
+  implicit val show: Show[ForkInfoEntries] = Show.show[ForkInfoEntries] { e =>
+    e.getEntries.mkString("[", ",", "]")
+  }
+
+  private case class ForkInfoEntriesImpl(maxSize: PosInt, value: Queue[ForkInfo]) extends ForkInfoEntries {
+    def add(entry: ForkInfo): ForkInfoEntries = {
+      val updated = (if (value.size === maxSize) value.tail else value) :+ entry
```

Comment:
So in case of reaching the size limit we cut the `head` and `append` the `entry`. It sounds like a combination of `dequeue` + `enqueue` of `FIFO` but you use neither of `Queue` specific methods.

So maybe instead of calling `tail` and `append` we can use the fact that both `dequeue` and `enqueue` have `O(1)` complexity which makes `O(1) + O(1) = O(1)`

```scala
val updated = if (value.size === maxSize) {
  value.dequeueOption
     .map { case (_, q) => q }
     .getOrElse(value)
     .enqueue(entry)
} else {
  value.enqueue(entry)
}
```

Reply:
Changed to

```
    def add(entry: ForkInfo): ForkInfoEntries = {
      val updated = if (value.size === maxSize) {
        value.dequeueOption.map { case (_, q) => q }
          .getOrElse(value)
          .enqueue(entry)
      } else {
        value.enqueue(entry)
      }

      copy(value = updated)
    }
```

Reply:
I can see the logic of using `enqueue` and `dequeue` but I think the implementation can be simplified to
```
if (value.size === maxSize) value.dequeue._2.enqueue(entry) else value.enqueue(entry)
```
since `maxSize` less than should not be permitted as it creates an unusable object (i.e. starting with an invalid state)

Reply:
Changed to:

```
    def add(entry: ForkInfo): ForkInfoEntries = {
      val updated =
        if (value.size === maxSize)
          value.dequeue._2.enqueue(entry)
        else
          value.enqueue(entry)

      copy(value = updated)
    }
```

Reply:
@sdavidp @cngo-github Ok I see. `value.dequeue` can throw `NoSuchElementException` for empty `Queue`, so I decided to go for `dequeueOption` but since `maxSize` is `PosLong` then it is secured by `value.size === maxSize` check. :+1: